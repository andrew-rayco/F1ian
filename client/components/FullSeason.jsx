import React from 'react'

import * as api from '../api'
import RaceOptions from './RaceOptions'

export default class FullSeason extends React.Component {
  constructor() {
    super()
    this.state = {
      races: null
    }
  }

  componentWillMount() {
    var location = window.location.pathname
    var pathArray = location.split('/')
    var thisSeason = pathArray[2]
    api.getRaces(thisSeason, (races) => {
      this.setState({races})
    })
  }

  listRaces(races) {
    const toggleStatus = {
      visibility: 'hidden'
    }
    return races.map((race) => {
      return (
        <div key={`key${race.raceId}`}>
          <RaceOptions key={race.raceId} props={race} visibility={toggleStatus} />
        </div>
      )
    })
  }

  render() {
    let racesInSeason = this.state.races
    console.log(JSON.stringify(racesInSeason))
    if (racesInSeason.length > 1) {
      console.log(racesInSeason[0].year)
      return (
        <div className="season">
          <h2>The {racesInSeason[0].year} Formula 1 Season</h2>
          {this.listRaces(racesInSeason)}
        </div>
      )
    } else {
      return (
        <div></div>
      )
    }
  }
}
