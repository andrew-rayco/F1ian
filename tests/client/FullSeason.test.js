import React from 'react'
import { shallow } from 'enzyme'
import FullSeason from '../../client/components/FullSeason'

let location = { pathname: "/season/2014" }

describe('FullSeason component', () => {
  let year = 1985
  const wrapper = shallow(<FullSeason location={location} />)

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('only displays once', () => {
    expect(wrapper).toHaveLength(1)
  })
})
