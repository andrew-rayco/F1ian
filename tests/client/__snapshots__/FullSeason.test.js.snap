// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullSeason component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FullSeason />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          The 
          2014
           Formula 1 Season
        </h2>,
        Array [
          <div>
            <RaceOptions
              props={
                Object {
                  "circuitId": 1,
                  "date": "2014-03-16",
                  "name": "Australian Grand Prix",
                  "raceId": 900,
                  "raceName": "Australian Grand Prix",
                  "raceUrl": "https://en.wikipedia.org/wiki/2014_Australian_Grand_Prix",
                  "round": 1,
                  "season-year": 2014,
                  "time": "06:00:00",
                  "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
                  "year": 2014,
                }
              }
              visibility={
                Object {
                  "visibility": "hidden",
                }
              }
            />
          </div>,
          <div>
            <RaceOptions
              props={
                Object {
                  "circuitId": 2,
                  "date": "2014-03-30",
                  "name": "Malaysian Grand Prix",
                  "raceId": 901,
                  "raceName": "Malaysian Grand Prix",
                  "raceUrl": "https://en.wikipedia.org/wiki/2014_Malaysian_Grand_Prix",
                  "round": 2,
                  "season-year": 2014,
                  "time": "08:00:00",
                  "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
                  "year": 2014,
                }
              }
              visibility={
                Object {
                  "visibility": "hidden",
                }
              }
            />
          </div>,
        ],
      ],
      "className": "season",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "The ",
            2014,
            " Formula 1 Season",
          ],
        },
        "ref": null,
        "rendered": Array [
          "The ",
          2014,
          " Formula 1 Season",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "key900",
        "nodeType": "host",
        "props": Object {
          "children": <RaceOptions
            props={
              Object {
                "circuitId": 1,
                "date": "2014-03-16",
                "name": "Australian Grand Prix",
                "raceId": 900,
                "raceName": "Australian Grand Prix",
                "raceUrl": "https://en.wikipedia.org/wiki/2014_Australian_Grand_Prix",
                "round": 1,
                "season-year": 2014,
                "time": "06:00:00",
                "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
                "year": 2014,
              }
            }
            visibility={
              Object {
                "visibility": "hidden",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "900",
          "nodeType": "class",
          "props": Object {
            "props": Object {
              "circuitId": 1,
              "date": "2014-03-16",
              "name": "Australian Grand Prix",
              "raceId": 900,
              "raceName": "Australian Grand Prix",
              "raceUrl": "https://en.wikipedia.org/wiki/2014_Australian_Grand_Prix",
              "round": 1,
              "season-year": 2014,
              "time": "06:00:00",
              "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
              "year": 2014,
            },
            "visibility": Object {
              "visibility": "hidden",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "key901",
        "nodeType": "host",
        "props": Object {
          "children": <RaceOptions
            props={
              Object {
                "circuitId": 2,
                "date": "2014-03-30",
                "name": "Malaysian Grand Prix",
                "raceId": 901,
                "raceName": "Malaysian Grand Prix",
                "raceUrl": "https://en.wikipedia.org/wiki/2014_Malaysian_Grand_Prix",
                "round": 2,
                "season-year": 2014,
                "time": "08:00:00",
                "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
                "year": 2014,
              }
            }
            visibility={
              Object {
                "visibility": "hidden",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "901",
          "nodeType": "class",
          "props": Object {
            "props": Object {
              "circuitId": 2,
              "date": "2014-03-30",
              "name": "Malaysian Grand Prix",
              "raceId": 901,
              "raceName": "Malaysian Grand Prix",
              "raceUrl": "https://en.wikipedia.org/wiki/2014_Malaysian_Grand_Prix",
              "round": 2,
              "season-year": 2014,
              "time": "08:00:00",
              "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
              "year": 2014,
            },
            "visibility": Object {
              "visibility": "hidden",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            The 
            2014
             Formula 1 Season
          </h2>,
          Array [
            <div>
              <RaceOptions
                props={
                  Object {
                    "circuitId": 1,
                    "date": "2014-03-16",
                    "name": "Australian Grand Prix",
                    "raceId": 900,
                    "raceName": "Australian Grand Prix",
                    "raceUrl": "https://en.wikipedia.org/wiki/2014_Australian_Grand_Prix",
                    "round": 1,
                    "season-year": 2014,
                    "time": "06:00:00",
                    "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
                    "year": 2014,
                  }
                }
                visibility={
                  Object {
                    "visibility": "hidden",
                  }
                }
              />
            </div>,
            <div>
              <RaceOptions
                props={
                  Object {
                    "circuitId": 2,
                    "date": "2014-03-30",
                    "name": "Malaysian Grand Prix",
                    "raceId": 901,
                    "raceName": "Malaysian Grand Prix",
                    "raceUrl": "https://en.wikipedia.org/wiki/2014_Malaysian_Grand_Prix",
                    "round": 2,
                    "season-year": 2014,
                    "time": "08:00:00",
                    "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
                    "year": 2014,
                  }
                }
                visibility={
                  Object {
                    "visibility": "hidden",
                  }
                }
              />
            </div>,
          ],
        ],
        "className": "season",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "The ",
              2014,
              " Formula 1 Season",
            ],
          },
          "ref": null,
          "rendered": Array [
            "The ",
            2014,
            " Formula 1 Season",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "key900",
          "nodeType": "host",
          "props": Object {
            "children": <RaceOptions
              props={
                Object {
                  "circuitId": 1,
                  "date": "2014-03-16",
                  "name": "Australian Grand Prix",
                  "raceId": 900,
                  "raceName": "Australian Grand Prix",
                  "raceUrl": "https://en.wikipedia.org/wiki/2014_Australian_Grand_Prix",
                  "round": 1,
                  "season-year": 2014,
                  "time": "06:00:00",
                  "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
                  "year": 2014,
                }
              }
              visibility={
                Object {
                  "visibility": "hidden",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "900",
            "nodeType": "class",
            "props": Object {
              "props": Object {
                "circuitId": 1,
                "date": "2014-03-16",
                "name": "Australian Grand Prix",
                "raceId": 900,
                "raceName": "Australian Grand Prix",
                "raceUrl": "https://en.wikipedia.org/wiki/2014_Australian_Grand_Prix",
                "round": 1,
                "season-year": 2014,
                "time": "06:00:00",
                "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
                "year": 2014,
              },
              "visibility": Object {
                "visibility": "hidden",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "key901",
          "nodeType": "host",
          "props": Object {
            "children": <RaceOptions
              props={
                Object {
                  "circuitId": 2,
                  "date": "2014-03-30",
                  "name": "Malaysian Grand Prix",
                  "raceId": 901,
                  "raceName": "Malaysian Grand Prix",
                  "raceUrl": "https://en.wikipedia.org/wiki/2014_Malaysian_Grand_Prix",
                  "round": 2,
                  "season-year": 2014,
                  "time": "08:00:00",
                  "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
                  "year": 2014,
                }
              }
              visibility={
                Object {
                  "visibility": "hidden",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "901",
            "nodeType": "class",
            "props": Object {
              "props": Object {
                "circuitId": 2,
                "date": "2014-03-30",
                "name": "Malaysian Grand Prix",
                "raceId": 901,
                "raceName": "Malaysian Grand Prix",
                "raceUrl": "https://en.wikipedia.org/wiki/2014_Malaysian_Grand_Prix",
                "round": 2,
                "season-year": 2014,
                "time": "08:00:00",
                "url": "http://en.wikipedia.org/wiki/2014_Formula_One_season",
                "year": 2014,
              },
              "visibility": Object {
                "visibility": "hidden",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
